src/tablet/tablet_impl.cc

```diff
@@ -3121,16 +3121,18 @@ int TabletImpl::LoadDiskTableInternal(uint32_t tid, uint32_t pid, const ::openml
         std::string manifest_file = snapshot_path + "MANIFEST";
         if (Snapshot::GetLocalManifest(manifest_file, manifest) == 0) {
             std::string snapshot_dir = snapshot_path + manifest.name();
-            PDLOG(INFO, "rename dir %s to %s. tid %u pid %u", snapshot_dir.c_str(), data_path.c_str(), tid, pid);
-            if (!::openmldb::base::Rename(snapshot_dir, data_path)) {
-                PDLOG(WARNING, "rename dir failed. tid %u pid %u path %s", tid, pid, snapshot_dir.c_str());
-                break;
-            }
-            if (unlink(manifest_file.c_str()) < 0) {
-                PDLOG(WARNING, "remove manifest failed. tid %u pid %u path %s", tid, pid, manifest_file.c_str());
-                break;
+            if (::openmldb::base::IsExists(snapshot_dir)) {
```

Comment:
why occurs that there is `MANIFEST` file but no snapshot dir

Reply:
one potential cause:
- re-start the tablet
- just after `rename` snapshot dir to `data`, but before `unlink` `MANIFEST`, stop tablet
- start the tablet: only `MANIFEST` exists, but the snapshot dir is not

Now the logic changes. We do a `hardlink` rather than `rename` in case some unexpected happens before following `MakeSnapshot`.